File: agent/tools/jira_get_issue.py
import asyncio
from typing import Any

from ..utils.jira import get_issue


def jira_get_issue(issue_key: str) -> dict[str, Any]:
    """Get a Jira issue by its key.

    Args:
        issue_key: The Jira issue key (e.g., "PROJ-123")

    Returns:
        Dictionary with 'issue' containing full issue details including:
        - key: Issue key (e.g., "PROJ-123")
        - fields: Issue fields including summary, description, status, assignee, etc.
        - error: Error message if the request failed
    """
    return asyncio.run(get_issue(issue_key))

File: src/tools/jiraGetIssue.ts
/**
 * jira_get_issue tool
 * Mirrors agent/tools/jira_get_issue.py
 */

import { tool } from "@langchain/core/tools";
import { z } from "zod";
import { getIssue } from "../utils/jira.js";

export const jiraGetIssue = tool(
  async ({ issueKey }) => {
    try {
      const issue = await getIssue(issueKey);
      return JSON.stringify({ success: true, issue });
    } catch (err) {
      return JSON.stringify({ error: String(err), status: "error" });
    }
  },
  {
    name: "jira_get_issue",
    description: "Get the details of a Jira issue by its key (e.g. 'PROJ-123').",
    schema: z.object({
      issueKey: z.string().describe("The Jira issue key, e.g. 'PROJ-123'."),
    }),
  },
);
